torch/csrc/distributed/c10d/comm.h

```diff
@@ -33,14 +41,30 @@ class GradBucket {
     return tensors_;
   }
 
+  const std::vector<size_t>& getOffsets() const {
+    return offsets_;
+  }
+
+  const std::vector<size_t>& getLengths() const {
+    return lengths_;
+  }
+
+  const std::vector<c10::IntArrayRef>& getSizesVec() const {
```

Comment:
What's the difference here between lengths and sizes, and why is one `size_t` type and the other is `IntArrayRef` type?

Reply:
`length` is the total 1D length like after flattening.
`sizes` is multi-dimensional shape info. Variable.sizes is similar to Tensor.shape.

Added a comment to describe the API.